conda_auth/cli.py

```diff
@@ -0,0 +1,102 @@
+from typing import NamedTuple
+
+import click
+from conda.base.context import context
+from conda.models.channel import Channel
+
+from .constants import OAUTH2_NAME, HTTP_BASIC_AUTH_NAME
+from .exceptions import CondaAuthError
+from .handlers import AuthManager, oauth2_manager, basic_auth_manager
+
+AUTH_MANAGER_MAPPING = {
+    OAUTH2_NAME: oauth2_manager,
+    HTTP_BASIC_AUTH_NAME: basic_auth_manager,
+}
+
+
+class ChannelData(NamedTuple):
+    """Used for providing commands with all the channel information they need"""
+
+    channel: Channel
+    settings: dict[str, str]
+    manager: AuthManager
+
+
+def get_example(channel):
+    """Function used to print a nice example for our users 😀"""
+    return (
+        "channel_settings:\n"
+        f"  - channel: {channel}\n"
+        "    auth: conda-auth-basic-auth\n"
+        "    username: user_one"
+    )
+
+
+def validate_channel(ctx, param, value):
+    """
+    Makes sure the channel exists in conda's configuration and returns a ``ChannelData`` object
+    if so.
+
+    TODO: This function is doing a lot more than a simple "validation"; Should be refactored or
+          renamed.
+    """
+    context.__init__()
```

Comment:
Is this to force the context initialization and configuration reading? Any risk in running it again in case it has already been executed elsewhere?

Reply:
I can remove this. I think this was a left over from something else I was doing earlier. Thanks for catching this.